### PR TITLE
Add windowRep sampling method flag -p

### PR DESCRIPTION
**Pull request by hanweny**
Minor fix sets.c where SetDelete won't update smallest element when the old smallest element is being deleted.

### CODE REVIEW

File: blant.c [hunk 1]
```diff
@@ -1527,25 +1656,35 @@ int RunBlantFromGraph(int k, int numSamples, GRAPH *G)
...
+            SampleGraphlet(G, V, Varray, k);
+            ProcessGraphlet(G, V, Varray, perm, g, k);
+        }
     }
-
-	if (_sampleMethod == SAMPLE_MCMC && !_windowRep)
+    if(_window)
+    {
+        for(i=0; i<_MAXnumWindowRep; i++)
+            free(_windowReps[i]);
```
Comment (HBBrennan) [inline blant.c]: Professor Hayes normally free's things right before exit in a special ```IF``` block.

File: blant.c [hunk 2]
```diff
@@ -1907,11 +2047,33 @@ int main(int argc, char *argv[])
...
 	    break;
-	case 'w':
-		_windowRep = true;
-		_windowSize = atoi(optarg); 
-		if(_windowSize < _k) Fatal("windowSize must be at least size k\n");
-		break;
+    case 'w':
+        _window = true;
+        _windowSize = atoi(optarg); 
+        if(_windowSize < _k) Fatal("windowSize must be at least size k\n");
```
Comment (HBBrennan) [inline blant.c]: I feel like this should error more gracefully if the window size cannot reasonably fit into memory.

File: blant.c [hunk 3]
```diff
@@ -1361,129 +1373,246 @@ void ProcessGraphlet(GRAPH *G, SET *V, unsigned Varray[], char perm[], TINY_GRAP
...
+            }
+            bitPos++;
+        }
     return Gint;
 }
-		
 
+int getMaximumIntNumber(int K) 
+{
+    assert(K >= 3 && K <= 8);
```
Comment (HBBrennan) [inline blant.c]: I would consider wrapping this in ``#IF PARANOID_ASSERTS``

File: blant.c (hunk 2 above)
Comment (waynebhayes) [reply blant.c]: Hmm.  If the graph can fit in memory why wouldn't the window?

Comment (hanweny) [reply blant.c]: Since I prealloc the 2D array for all combinations (W choose K), it's the 2D table that might not fit into the memory

**Pull request merged**